testdata/rabbit/webScraping/main.go

```diff
@@ -0,0 +1,92 @@
+package main
+
+import (
+	"encoding/csv"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"os"
+)
+
+func main() {
+	links := readCSV()
+	if links == nil {
+		log.Fatalf("Failed to find any links.")
+	}
+
+	for _, link := range links {
+		getDataFromWeb(link)
+	}
+
+}
+
+func readCSV() []string {
+	filePath := "./top500Domains.csv"
+	targetColumn := "Root Domain"
+
+	file, err := os.Open(filePath)
+	if err != nil {
+		log.Fatalf("Failed to open CSV file: ", err)
+	}
+
+	defer file.Close()
+
+	// Create a new CSV reader reading from the opened file
+	reader := csv.NewReader(file)
+
+	// Read the CSV records
+	records, err := reader.ReadAll()
+	if err != nil {
+		log.Fatalf("Failed to read CSV records: ", err)
+	}
+
+	header := records[0]
+	// Find the index of the target column
+	targetColumnIndex := -1
+	for i, column := range header {
+		if column == targetColumn {
+			targetColumnIndex = i
+			break
+		}
+	}
+
+	if targetColumnIndex == -1 {
+		log.Fatalf("Target column not found: ", targetColumn)
+	}
+
+	links := make([]string, len(records)-1)
+	for i, record := range records[1:] {
+		// add a valid protocol scheme "https://"
+		links[i] = "https://" + record[targetColumnIndex]
+	}
+	return links
+}
+
+func getDataFromWeb(webURL string) {
+	resp, err := http.Get(webURL)
+	if err != nil {
+		log.Printf("Failed to fetch %s: %s\n", webURL, err)
+		return
+	}
+
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		log.Println("Server returned non-200 status code: ", resp.StatusCode)
+		return
+	}
+
+	body, err := io.ReadAll(resp.Body)
```

Comment:
In general, be careful of ReadAll(). That function should not be used on inputs that you don't control because some bad person might give you an infinite stream of data.

A simple fix is to use a LimitReader *and then* to use ReadAll of that.  This guarantees you won't read more than N bytes.